packages/core/src/events/input/load/vectorstore/base.ts

```diff
@@ -0,0 +1,48 @@
+import { Serializable } from '../../../../load/serializable.js';
+import { Context } from '../docs/context.js';
+
+export interface BaseVectorStoreField {}
+
+export abstract class BaseVectorStore
+  extends Serializable
+  implements BaseVectorStoreField
+{
+  declare FilterType: object | string;
+
+  _isSerializable = false;
+
+  _namespace: string[] = [
+    'events',
+    'input',
+    'load',
+    'vectorstore',
+    this._vectorstoreType(),
+  ];
+
+  constructor(fields?: Partial<BaseVectorStoreField>) {
+    super(fields);
+  }
+
+  abstract _vectorstoreType(): string;
+
+  abstract addVectors(
```

Comment:
shall we also have a add document method, so that the embed method can be called within?

Reply:
cool, thanks for clarification